### ISSUE TITLE
[Feature Request] TensorRT support

### ISSUE THREAD

**Issue by BlueAccords** [comfyanonymous/ComfyUI#29]
I think I remember somewhere you were looking into supporting tensortRT models, is that still in the backlog somewhere? or would implementing support for tensortRT require too much rework of the existing codebase?

**Comment by comfyanonymous** [comment]
Yes it's planned. There's a few more things I want to implement before. Unless something better comes out in the meantime I'm going to implement it.

**Comment by 78Alpha** [comment]
Be prepared to pull your hair out if you get to it. Getting it setup on one machine is a chore, but getting it to work on someone else's is worse. Many TensorRT implementations have fallen because of it. Volta-ML and SDA-node are the larger examples.

**Comment by CyberTimon** [comment]
I would love to see tensorrt support, because SD XL is quite slow.
Also, tensorrt only seems to support max 768x768px. Do you think it is somehow possible to pass the SDXL 1024 in?

**Comment by 78Alpha** [comment]
There was a pull for automatic that references the limits of TRT. 

https://github.com/AUTOMATIC1111/stable-diffusion-webui-tensorrt/pull/36

There is an upper limit of what it can do. As an example, if you have your batch size set to 8, you may not be able to generate dynamic images greater than 512x480 and the like.

I took note of some issues I've found during the year of TRT implementations flooding GitHub.

https://github.com/AUTOMATIC1111/stable-diffusion-webui-tensorrt/issues/46#issuecomment-1644403272

**Comment by comfyanonymous** [comment]
Yeah I'm going to go with AITemplate instead of TRT unless they add a way to replace the weights at runtime.

**Comment by CyberTimon** [comment]
Yeah AiTemplate seems good too.

**Comment by nistvan86** [comment]
SDXL models now have a TensorRT variant. https://huggingface.co/stabilityai/stable-diffusion-xl-1.0-tensorrt

**Comment by al-swaiti** [comment]
yeah its there! any idea how make it work in comfyui

**Comment by Erehr** [comment]
Aaaand new drivers just dropped promising 2X performance boost with TRT
[https://www.nvidia.com/en-us/geforce/news/game-ready-driver-dlss-3-naraka-vermintide-rtx-vsr/](https://www.nvidia.com/en-us/geforce/news/game-ready-driver-dlss-3-naraka-vermintide-rtx-vsr/)

**Comment by BassJMagan** [comment]
> Aaaand new drivers just dropped promising 2X performance boost with TRT https://www.nvidia.com/en-us/geforce/news/game-ready-driver-dlss-3-naraka-vermintide-rtx-vsr/

and with that I'd love to see the fruits of this 2x speed boost as that could greatly improve my workflow on my 3090.

**Comment by xueqing0622** [comment]
support!
https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT

**Comment by SplitMilky** [comment]
Ok so how do i translate that support instructions for auto1111 to compfy.  As I was using auto1111 but I was getting issues creating and using the optimized versions

**Comment by boricuapab** [comment]
Bumping this, would be great if there's a way to use locally converted sdxl trt models in comfy to reap the benefits of the speed boost.  The 7s/it was out of the norm as I was browsing civit, but the results Im gettings are 1.6s/it for 1024x1024 res.

![trtBirdy](https://github.com/comfyanonymous/ComfyUI/assets/61924816/958b9cba-fe98-41c6-92c6-4df0d6b746bf)

**Comment by phineas-pta** [comment]
advanced users can try my node: https://github.com/phineas-pta/comfy-trt-test

**Comment by Dunc4n1dah0** [comment]
I just tested Nvidia's A1111 TRT extension and results are 2x faster indeed (at least for 512x768 simple generation).

**Comment by robinjhuang** [comment]
@BlueAccords Have you tried this?
https://github.com/comfyanonymous/ComfyUI_TensorRT

**Comment by maxpaynestory** [comment]
I tried ComfyUI_TensorRT and the results were slower. I have RTX 3050 Ti laptop GPU.

Pytorch=6.01s/it
TensorRT=9.63s/it

![Image](https://github.com/user-attachments/assets/66b938e8-cb77-420e-9834-adac51985e23)
